microsoft/iot-cardboard-js
# Data History Spike - migrate local storage to new keys for environment data

### Summary of changes 🔍 
Here is the design doc for data history widget including the ADO item: [Data History Widget.doc](https://microsoft.sharepoint.com/:w:/r/teams/Azure_IoT/IoTPlat/Shared%20Documents/Digital%20Twins/36.%20Digital%20Twins%20-%20ExACT%20Team/3D%20Scenes/DevDocs%20(Notion%20cont.)/Feature%20reviews/Data%20history%20widget.docx?d=w6e73e5a305f44fa6a0579ed5c828273a&csf=1&web=1&e=V3xjGO)
- Storing resource id in the localStorage and deeplink to easily retrieve ADX connection information of the selected ADT instance which will be used as the default "connectionString" for data history widget (See the design doc for details about changes).
- Made many name changes to make variables explicit and fixed some types, that is why there are many changes across files.
- Moved some methods and their unit tests which are used across multiple components to Utils.ts file.

![image](https://user-images.githubusercontent.com/45217314/193365175-6aad3955-58b2-4713-bd0d-618466ceaef3.png)

### Testing 🧪
1. Use ADT3DScenePage component
2. On page load, open localstorage and see if the migration of localstorage variables to new key of "**cb-environment-configuration**" and "**cb-environment-options**" successful and persistent
3. See if the old local storage keys of "cb-environments", "cb-containers", "cb-storage-accounts", "cb-selected-environment",  "cb-selected-container" are removed (we kept the keys which are passed as prop to the EnvironmentPicker thinking it might be used by library consumers in their app like ADT Explorer, only removed cardboard defined localstorage keys)
4. On EnvironmentPicker save, see if "cb-environment-options" (by EnvironmentPicker) and "cb-environment-configuration" (by DeeplinkContext) is getting updated.
5. Click on share to build a deeplink and copy the link in a new tab and see if the resource id is exposed in the link
6. See you localstorage is getting updated after going to deeplink in that new tab (Step-2).

Or, your can clear localStorage altogether (or specifically remove "cb-" prefixed variables since this PR changed the data structure that we keep in localstorage for selected environment items)


### Checklist ✔️
